### PR TITLE
Added provision to edit application if checksum is present in URL

### PR DESCRIPTION
**Pull request by Edzelopez**

### CODE REVIEW

File: CRM/Aoservicelisting/Form/ProviderApplication.php [hunk 1]
```diff
@@ -34,8 +34,33 @@ class CRM_Aoservicelisting_Form_ProviderApplication extends CRM_Core_Form {
...
+    // Validate checksum only if specified in the URL, for non-logged in users.
+    if (!empty($cid) && !empty($cs)) {
+      if (!CRM_Contact_BAO_Contact_Utils::validChecksum($cid, $cs)) {
+        $invalidUser = TRUE;
+      }
+      else {
+        // This is a valid user.
+        $loggedInContactId = $cid;
+      }
+    }
```
Comment (monishdeb) [inline CRM/Aoservicelisting/Form/ProviderApplication.php:56]: @Edzelopez if the $cid is not provided in link then we should consider the loggedIncontactId right? 

like here it should be 
```
...
elseif($loggedInContactId && empty($cid && !empty($cs)) {
   $cid = $loggedInContactId;
}
```
right ?

Comment (Edzelopez) [reply CRM/Aoservicelisting/Form/ProviderApplication.php:56]: Not quite. If the $cid is not provided in the link, the service listing application should be treated as a create form. On L38 above, we are already checking the fact that the user is logged in, and so a logged in user will have no reason to have $cid in the link.
If we add the else part, L58 below will consider the form to be submitted by an invalid user.

Comment (monishdeb) [reply CRM/Aoservicelisting/Form/ProviderApplication.php:56]: agree.

**Comment by monishdeb** [comment]
Tested on jma.st and works fine.

**Pull request merged**